### PR TITLE
Added support for multiple Figma teams

### PR DESCRIPTION
**Pull request by GLaDO8**
## Description

1. Added support for multiple Figma teams. Multiple teams can be added by providing a comma separated list of team IDs to the extension preferences. 
2. Better caching support through Raycast hooks

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/figma-files/README.md
Comment (michaelwschultz) [inline extensions/figma-files/README.md:1]: Thanks so much for updating this no nicely.

File: extensions/figma-files/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # Figma Files Changelog
 
+## [Multi-team support and better caching] - 2023-05-12
+
+- Added support for multiple teams. You can now search across multiple teams by providing a comma separated list of teamIDs in preferences.
+- Search results are cached for smoother experience. You can now also search for files when offline.
```
Comment (michaelwschultz) [inline extensions/figma-files/CHANGELOG.md:6]: 🎉

File: extensions/figma-files/src/hooks/useVisitedFiles.ts [hunk 2]
```diff
@@ -25,7 +26,7 @@ export async function clearVisitedFiles() {
 }
 
 export function useVisitedFiles() {
-  const [files, setFiles] = useState<File[]>();
+  const [files, setFiles] = useCachedState<File[]>("visitedFiles");
```
Comment (michaelwschultz) [inline extensions/figma-files/src/hooks/useVisitedFiles.ts:29]: Rad.

File: extensions/figma-files/src/index.tsx [hunk 3]
```diff
@@ -2,14 +2,25 @@ import { Application, getApplications, Grid } from "@raycast/api";
...
 import { ErrorView } from "./components/ErrorView";
 import { useVisitedFiles } from "./hooks/useVisitedFiles";
-import { useProjectFiles } from "./hooks/useProjectFiles";
+import { resolveAllFiles } from "./components/fetchFigmaData";
 import { useEffect, useState } from "react";
+import { useCachedPromise } from "@raycast/utils";
 
 export default function Command() {
-  const { projectFiles, isLoading: isLoadingProjectFiles, hasError } = useProjectFiles();
+  const { data, isLoading, error } = useCachedPromise(
```
Comment (michaelwschultz) [inline extensions/figma-files/src/index.tsx:10]: Thanks for updating!

**Review by michaelwschultz — APPROVED** [review]
This looks fantastic @GLaDO8! Multiple teams has been such a highly requested feature but I just haven't had time to implement it. You're approach here is great and love the clean up/optimizations as well. Thanks again!

File: extensions/figma-files/package.json [hunk 4]
```diff
@@ -39,12 +40,13 @@
       "name": "TEAM_ID",
       "type": "password",
```
Comment (pernielsentikaer) [inline extensions/figma-files/package.json:41]: Shouldn't this be a text field

Comment (GLaDO8) [reply extensions/figma-files/package.json:41]: Ah yes. Let me change this.

**Pull request merged**